mono/mini/driver.c

```diff
@@ -1745,6 +1757,8 @@ mono_jit_parse_options (int argc, char * argv[])
 			mono_counters_enable (-1);
 			mono_atomic_store_bool (&mono_stats.enabled, TRUE);
 			mono_atomic_store_bool (&mono_jit_stats.enabled, TRUE);
+		} else if (strncmp (argv [i], "--stats=", 8) == 0) {
+			handle_stats_name (argv [i]);
```

Comment:
If it hits this condition, it means it has *not* run the block for `strcmp (argv [i], "--stats") == 0`. While you will set the method name, you won't enable the counters or stat collection.